release/dist/qnap/pkgs.go

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package qnap contains dist Targets for building QNAP Tailscale packages.
+//
+// QNAP dev docs over at https://www.qnap.com/en/how-to/tutorial/article/qpkg-development-guidelines.
+package qnap
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+
+	"tailscale.com/release/dist"
+)
+
+type target struct {
+	goenv  map[string]string
+	arch   string
+	signer dist.Signer
+}
+
+func (t *target) String() string {
+	return fmt.Sprintf("qnap/%s", t.arch)
+}
+
+func (t *target) Build(b *dist.Build) ([]string, error) {
+	// Stop early if we don't have docker running.
+	if _, err := exec.LookPath("docker"); err != nil {
+		return nil, fmt.Errorf("docker not found, cannot build: %w", err)
+	}
+
+	qnapBuilds := getQnapBuilds(b)
+	inner, err := qnapBuilds.buildInnerPackage(b, t.goenv)
```

Comment:
I don't fully understand why we need the `qnapBuilds` type. IIUC, we could build the binaries and package them up with docker from this same function?

Reply:
The nice thing about `qnapBuilds` is that it memoizes the built binaries so we only have to build them once per `arch`.

Reply:
aha, makes sense!
Go's build cache would probably keep it fast anyway, but this works too.
could you add some doc comments on `qnapBuilds` to explain its purpose?

Reply:
Yes, good idea, will add docs!